curriculum/challenges/english/blocks/workshop-shape-manager/69159948e52168265730f74f.md

```diff
@@ -0,0 +1,422 @@
+---
+id: 69159948e52168265730f74f
+title: Step 25
+challengeType: 25
+dashedName: step-25
+---
+
+# --description--
+
+You need a way to safely access the select element and guard against `null` values. For that, create a `Val` constant with the value `e.currentTarget` and type it `HTMLSelectElement` using the `as` keyword.
+
+Under the constant, use an `if` statement to check if `Val` is falsy by negating it. In the `if` block, return `target value not found`.
+
+# --hints--
+
+You should have a constant called `Val`.
+
+```js
+assert.match(code, /const\s+Val/)
+```
+
+Your `Val` constant should have a `e.currentTarget` value typed `HTMLSelectElement` using the `as` keyword.
+
+```js
+assert.match(code, /const\s+Val\s*=\s*e\.currentTarget\s+as\s+HTMLSelectElement;?/)
+```
+
+You should have an `if` statement that checks if `Val` is null.
+
+```js
+assert.match(handleShapeSelect.toString(), /if\s*\((!Val|Val===null)\)\s*\{\s*/)
+```
+
+Your `if` statement should return `target value not found`.
+
+```js
+assert.match(handleShapeSelect.toString(), /if\s*\((\s*!Val\s*)\)\s*\{\s*return\s*("|')target\s+value\s+not\s+found("|');?\s*\}/)
```

Comment:
This code does not pass because curly braces are required by the test:
```ts
const Val = e.currentTarget as HTMLSelectElement
if (!Val) return 'target value not found'
```